### ISSUE TITLE
yield lambda branch coverage

### ISSUE THREAD

**Issue by nedbat** [coveragepy/coveragepy#308]
*Originally reported by* **rdn_ (Bitbucket: [rdn_](https://bitbucket.org/rdn_), GitHub: Unknown)**

----------------------------------------

Version 3.7.1

`coverage run --branch x.py`

Missing branch:

```python
def run():
    for i in range(10):
        yield lambda: i  # partial coverage missing jump to previous line

for f in run():
    print f()
```

Missing branch # 2:

```python
def run():
    yield lambda: 100  # partial coverage missing jump to next line
    for i in range(10):
        yield lambda: i  # partial coverage missing jump to previous line

for f in run():
    print f()
```

Fully covered:

```python
def run():
    yield lambda: 100  # no branch miss

for f in run():
    print f()
```

Wasn't sure if I was overlooking something, but given the 2nd case I think the problem is in coverage.py

----------------------------------------
- Bitbucket: https://bitbucket.org/ned/coveragepy/issue/308

**Comment by nedbat** [comment]
Fixed in [2e992bf44abaf0d3c78134a4eb9724f015d23e00 (bb)](https://bitbucket.org/ned/coveragepy/commits/2e992bf44abaf0d3c78134a4eb9724f015d23e00)

**Comment by nedbat** [comment]
Issue #402 was marked as a duplicate of this issue.
